### PR TITLE
ManagedIdentityClientAssertion: Set clientId for both ManagedIdentity and WorkloadIdentity

### PR DESCRIPTION
**Pull request by sebader**
# ManagedIdentityClientAssertion: Set clientId for both ManagedIdentity and WorkloadIdentity

- [x] You've read the [Contributor Guide](https://github.com/AzureAD/microsoft-identity-web/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/AzureAD/microsoft-identity-web/blob/master/CODE_OF_CONDUCT.md).
- [ ] You've included unit or integration tests for your change, where applicable.
- [x] You've included inline docs for your change, where applicable.
- [x] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.

Summary of the changes (Less than 80 chars)

## Description

See issue below

Fixes https://github.com/AzureAD/microsoft-identity-web/issues/2643

Kind of clashes with https://github.com/AzureAD/microsoft-identity-web/issues/2644 but this will work at least as a short-term fix for people wanting to use WorkloadIdentity (on AKS)

## Alternative

An alternative approach would be a completely new class dedicated for WorkloadIdentity:

```csharp
public class WorkloadIdentityClientAssertion : ClientAssertionProviderBase
{
    private readonly TokenCredential _credential;

    /// <summary>
    /// See https://aka.ms/ms-id-web/certificateless.
    /// </summary>
    /// <param name="managedIdentityClientId">ClientId of the managed identity, in case multiple identities are federated with the service account</param>
    public WorkloadIdentityClientAssertion(string? managedIdentityClientId)
    {
        _credential = new WorkloadIdentityCredential(
            new WorkloadIdentityCredentialOptions
            {
                ClientId = managedIdentityClientId
            });
    }

    /// <summary>
    /// Prototype of certificate-less authentication using a signed assertion
    /// acquired with managed identity (certificateless).
    /// </summary>
    /// <returns>The signed assertion.</returns>
    protected override async Task<ClientAssertion> GetClientAssertion(CancellationToken cancellationToken)
    {
        var result = await _credential.GetTokenAsync(
            new TokenRequestContext(["api://AzureADTokenExchange/.default"], null),
            cancellationToken).ConfigureAwait(false);
        return new ClientAssertion(result.Token, result.ExpiresOn);
    }
}
```

### CODE REVIEW

**Comment by jennyf19** [comment]
@bgavrilMS can you review? Is this going in the right direction for where you want this to go?

File: src/Microsoft.Identity.Web.Certificateless/ManagedIdentityClientAssertion.cs [hunk 1]
```diff
@@ -13,28 +13,31 @@ namespace Microsoft.Identity.Web
...
-        /// <param name="managedIdentityClientId"></param>
+        /// <param name="managedIdentityClientId">Optional ClientId of the Managed Identity or Workload Identity</param>
         public ManagedIdentityClientAssertion(string? managedIdentityClientId)
         {
-            _managedIdentityClientId = managedIdentityClientId;
+            _credential = new DefaultAzureCredential(
+                new DefaultAzureCredentialOptions
+                {
+                    ManagedIdentityClientId = managedIdentityClientId,
+                    WorkloadIdentityClientId = managedIdentityClientId
```
Comment (jmprieur) [inline src/Microsoft.Identity.Web.Certificateless/ManagedIdentityClientAssertion.cs:28]: Let's use the best of both suggestions (@psignoret and yours):
- let's use both the ManagedIdentity and WorkloadIdentity credentials
- but let's disable the credentials that don't make sense when Managed identity is expected

```suggestion
                    WorkloadIdentityClientId = managedIdentityClientId,
                    ExcludeAzureCliCredential = true,
                    ExcludeAzureDeveloperCliCredential = true
                    ExcludeAzurePowerShellCredential = true,
                    ExcludeInteractiveBrowserCredential = true,
                    ExcludeSharedTokenCacheCredential = true,
                    ExcludeVisualStudioCodeCredential = true,
                    ExcludeVisualStudioCredential = true
```

**Review by jmprieur — APPROVED** [review]
LGTM with the suggestion which would also fix https://github.com/AzureAD/microsoft-identity-web/issues/2644

**Review by jennyf19 — APPROVED** [review]
:shipit:

File: src/Microsoft.Identity.Web.Certificateless/ManagedIdentityClientAssertion.cs [hunk 2]
```diff
@@ -13,28 +13,38 @@ namespace Microsoft.Identity.Web
...
         public ManagedIdentityClientAssertion(string? managedIdentityClientId)
         {
-            _managedIdentityClientId = managedIdentityClientId;
+            _credential = new DefaultAzureCredential(
+                new DefaultAzureCredentialOptions
+                {
+                    ManagedIdentityClientId = managedIdentityClientId,
+                    WorkloadIdentityClientId = managedIdentityClientId,
+                    ExcludeAzureCliCredential = true,
+                    ExcludeAzureDeveloperCliCredential = true
```
Comment (jmprieur) [inline src/Microsoft.Identity.Web.Certificateless/ManagedIdentityClientAssertion.cs:30]:
```suggestion
                    ExcludeAzureDeveloperCliCredential = true,
```

**Comment by sebader** [comment]
Thanks for the addition!

**Pull request merged**